src/hotspot/share/opto/superword.hpp

```diff
@@ -57,16 +57,281 @@
 
 class VPointer;
 
+// The PairSet is a set of pairs. These are later combined to packs,
+// and stored in the PackSet.
+class PairSet : public StackObj {
+private:
+  const VLoop& _vloop;
+  const VLoopBody& _body;
+
+  // Doubly-linked pairs. If not linked: -1
+  GrowableArray<int> _left_to_right; // bb_idx -> bb_idx
+  GrowableArray<int> _right_to_left; // bb_idx -> bb_idx
+
+  // List of all left elements bb_idx, in the order of pair addition.
+  GrowableArray<int> _lefts_in_insertion_order;
```

Comment:
Note: It turns out that the pair-extension is quite sensitive to the order in which pairs are found. Keeping the insertion order does something similar to a BFS. If this order is changed, then some of our examples don't pack properly. Well, they pack correctly on a local level, but then that prevents the proper packing further on.
Maybe I can remove the need for this order in the future, but for now I'll keep the order, the same order as the `_packset`'s pairs had.